chevah/brink
# [#3237] Publish tag on release and lint the release notes.

Scope
=====

This update the release code to push the tag on release.

It also add lint rules for release notes.


Changes
=======

Add a '.ignore' release notes fragment category which is not part of the release notes.
We still need to create that file for each branch to make sure that we don't forget to add the release notes.... we will see if this is helpful or annoying :)

Update `rqm` code to look at target branch based on the PR, to check PR test run status and to publish tags on release.


How to test
=========

reviewers: @alibotean  

For the lint part, make sure there is no release notes for this branch and run lint. It should complain that this branch has no release fragments.

Rename the branch to include '-release-' into its name and run paver lint. If it has release fragments it should  complain that release branches should have no release fragments.

------

for the release part, we would need to do a test release to check that all is ok... and for now we don't have brink set up for release... so this needs to be tested on server...ugly stuff... I need to create a release builder for brink.

For now I have manually called these command in a similar way to the buildbot does it... but it will messed your local repo as buildbot uses a bare repo

```
# pre-commit
$ GITHUB_TOKEN=12345 GITHUB_PULL_ID=78 paver merge_init

# No release merge
$ GITHUB_TOKEN=12345 GITHUB_PULL_ID=78 TEST_AUTHOR='io' RQM=no paver merge_commit

# Merge for release
$ GITHUB_TOKEN=12345 GITHUB_PULL_ID=78 TEST_AUTHOR='io' RQM=yes paver merge_commit
```
